android/app/src/main/kotlin/com/jabook/app/jabook/audio/AudioPlayerService.kt

```
@@ -510,6 +542,24 @@ public class AudioPlayerService : MediaLibraryService() {
             bookLoudnessCompensator.applyCompensation(bookId, booksDao, playerServiceScope) { getActivePlayer() }
         }
 
+        // Crossfade to a different book if crossfade is enabled and currently playing
+        if (groupPath != null &&
+            groupPath != currentGroupPath &&
+            playerConfigurator?.audioProcessingSettings?.isCrossfadeEnabled == true &&
+            isPlaying &&
+            crossFadePlayer != null
```

detekt (1.23.8)

[warning] This condition is too complex (5). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)